### PR TITLE
f8801 for 2019

### PR DESCRIPTION
**Pull request by redstreet**

### CODE REVIEW

File: 2018/f8801_2019.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            f['6'] = f6251.EXEMPT_LIMITS[inputs['status']]
+            f['7'] = max(0, f['4'] - f['6'])
+            f['8'] = f['7'] * 0.25
+            f['9'] = max(0, f['5'] - f['8'])
+            f['10'] = max(0, f['4'] - f['9'])
+            if f['10'] == 0:
+                f['15'] = 0
+            else:
+                # TODO: form 2555
+                if (f1040['13'] and not sched_d.mustFile()) or f1040['9b'] or \
```
Comment (davidcmoore) [inline 2018/f8801_2019.py]: '13' should be 's13' and '9b' should be '3a'

File: 2018/f8801_2019.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    f['31'] = min(f['27'], f['30'])
+                    f['32'] = f['27'] - f['31']
+                    f['33'] = f6251.amt(inputs['status'], f['32'])
+                    f['34'] = f1040.BRACKET_LIMITS[inputs['status']][1]
+                    f['35'] = cg_worksheet['7']
+                    f['36'] = max(0, f['34'] - f['35'])
+                    f['37'] = min(f['27'], f['28'])
+                    f['38'] = min(f['36'], f['37'])
+                    f['39'] = f['37'] - f['38']
+                    f['40'] = f1040.BRACKET_LIMITS[inputs['status']][-1]
```
Comment (davidcmoore) [inline 2018/f8801_2019.py]: BRACKET_LIMITS should be replaced with CAPGAIN20_LIMITS on this line.

File: 2019/f8801.py [hunk 3]
```diff
@@ -10,8 +10,8 @@ def __init__(f, inputs, f1040, f6251):
             return
 
         f.must_file = True
-        f['22'] = max(0, f1040['11a'] + f1040['s46'] - \
-            (f1040.rowsum(['12a', 's48', 's49', 's50', 's51', 's53', 's54']) or 0))
+        f['22'] = max(0, f1040['12a'] + f1040['s2_2'] - \
+            (f1040.rowsum(['13a', 's3_1', 's3_2', 's3_3', 's3_4', 's3_5', 's3_6']) or 0))
         f['23'] = f6251.get('9')
         f['24'] = max(0, f['22'] - f['23'])
         f.comment['25'] = 'Minimum tax credit'
```
Comment (davidcmoore) [inline 2019/f8801.py]: Line 21 should say "2020"

File: 2019/f8801.py [hunk 4]
```diff
@@ -10,8 +10,8 @@ def __init__(f, inputs, f1040, f6251):
             return
```
Comment (davidcmoore) [inline 2019/f8801.py]: Delete the "NOT FINAL" comment from the top of the file now that's you've finished the work.

File: 2019/f8801.py (hunk 3 above)
Comment (redstreet) [reply 2019/f8801.py]: done

File: 2018/f8801_2019.py (hunk 1 above)
Comment (redstreet) [reply 2018/f8801_2019.py]: good catch, done.

File: 2018/f8801_2019.py (hunk 2 above)
Comment (redstreet) [reply 2018/f8801_2019.py]: Hmm,
1) I'm not seeing exactly how this works. These refer to status on 1040-NR, right? If you understand this better than me, it'd be great if you could confirm what it should be.
2) I copied this from 2017/f8801_2018.py. That one uses BRACKET_LIMITS. Does that need to be fixed too?

Comment (davidcmoore) [reply 2018/f8801_2019.py]: inputs['status'] is just your filing status.

In 2017, these thresholds were pegged to the regular tax brackets, but in 2018 with the tax law changes, the capital gain thresholds are now called out separately. You can see the same difference if you look at 2017/f6251.py versus 2018/f6251.py.

Anyways, f['34'] should be f1040.CAPGAIN15_LIMITS[inputs['status']] and f['40'] should be f1040.CAPGAIN20_LIMITS[inputs['status']]

Comment (redstreet) [reply 2018/f8801_2019.py]: Thanks, done.

**Pull request merged**